### PR TITLE
Track processing progress. Part 2

### PR DESCRIPTION
**Pull request by MichaelKovalchuk**
File uploads are now processed in batches. 
The queue was added to control the order
Max concurrency is set to 5
Max uploads limit is not necessary for now

We will always show newer bulk uploads on the top
The files are getting processed on the first-come first-serve basis

### CODE REVIEW

File: src/state/uploader.js [hunk 1]
```diff
@@ -17,133 +17,172 @@ const sleep = async(delay = 0) => {
...
-				yield sleep(randomizeDelay(5000, 1000));
-				yield * monitorProgress(content, revision, progressCallback);
-			}.bind(this);
+			const uploadInfo = { file, progress: 0, extension: file.name.split('.').pop(), err: null, batch };
+			let count = 0;
+			this.uploads.forEach(ui => {
+				if (ui.batch === batch) {
+					count += 1;
+				}});
+			this.uploads.splice(count, 0, uploadInfo);
```
Comment (MichaelKovalchuk) [inline src/state/uploader.js]: this is to control order in which the items will be shown in the UI. 
e.g. Each batch should appear at the top but all executions should be on the first-come first-serve basis

File: src/state/uploader.js [hunk 2]
```diff
@@ -17,133 +17,173 @@ const sleep = async(delay = 0) => {
...
+
+	await sleep(randomizeDelay(5000, 1000));
+	await this.monitorProgress(content, revision, progressCallback);
+	/* eslint-enable no-invalid-this */
+}
+
+async function uploadWorkflow({ file, extension }) {
+	/* eslint-disable no-invalid-this */
+	try {
+		this.runningJobs += 1;
```
Comment (qwangd2l) [inline src/state/uploader.js]: Are we missing incrementing `uploadsInProgress` or can we just use `runningJobs`?
Seems like runningJobs does the same thing but the status header title right now depends on `uploadsInProgress`'s count.

Comment (MichaelKovalchuk) [reply src/state/uploader.js]: Nice catch. I forgot to include it. we still need uploadsInProgress

Comment (MichaelKovalchuk) [reply src/state/uploader.js]: uploadsInProgress is a total number of jobs still scheduled to be processed and runningJobs is a number of jobs actually being processed.

File: src/state/uploader.js [hunk 3]
```diff
@@ -17,133 +17,174 @@ const sleep = async(delay = 0) => {
...
+				revisionId: revision.id,
+				error: err
+			});
+			try {
+				await this.apiClient.deleteRevision({
+					contentId: content.id,
+					revisionId: revision.id
+				});
+			} catch (error) {
+				// Catch the error to delete here so that it doesn't fall through
```
[the remark this answers is not in the document]
Comment (MichaelKovalchuk) [reply src/state/uploader.js]: for now, yes. this is for deletion of the content that did not finish processing. if the revision was not created yet - we should ignore it, etc.

[the remark this answers is not in the document]
Comment (MichaelKovalchuk) [reply src/state/uploader.js]: the error will still show, but the original one - the one

**Pull request merged**